### PR TITLE
Add XML and text regression tests for crm_mon

### PR DESCRIPTION
**Pull request by clumens**

### CODE REVIEW

File: tools/crm_mon_print.c [hunk 1]
```diff
@@ -208,7 +208,7 @@ print_resources(pcmk__output_t *out, pe_working_set_t *data_set,
             continue;
         }
 
-        if (!pe__rsc_running_on_any_node_in_list(rsc, only_show)) {
+        if (pcmk__rsc_is_filtered(rsc, only_show)) {
```
Comment (kgaillot) [inline tools/crm_mon_print.c:211]: I think there's one case where this shouldn't apply, if we're showing inactive resources and there's no host filter, then we should show all inactive resources regardless of whether they're allowed or known anywhere. I'm not sure how that would fit in with the other callers.

In other words, the current definition of pe__rsc_allowed_on_any_node_in_list() makes sense based on its name, but crm_mon needs that one case to be different.

Comment (clumens) [reply tools/crm_mon_print.c:211]: I've got a patch that should address this, but strangely it doesn't seem to make any difference in the test case output.  It seems like either this is already being handled in the code, or I don't have a test case that is showing the problem.

Comment (kgaillot) [reply tools/crm_mon_print.c:211]: It would be a corner case, primarily at cluster start-up, at resource creation, or immediately after a refresh, before the resource has been probed or started. Deleting all lrm_rsc_op entries for the resource should simulate it.

Comment (kgaillot) [reply tools/crm_mon_print.c:211]: Oh and it would also need target-role=Stopped or be banned from every node to not show up in allowed nodes

Comment (clumens) [reply tools/crm_mon_print.c:211]: Okay, there we go.  I think I see the problem (and the fix) now.

File: tools/crm_mon_print.c [hunk 2]
```diff
@@ -206,9 +206,8 @@ print_resources(pcmk__output_t *out, pe_working_set_t *data_set,
          */
         } else if (!partially_active && is_not_set(mon_ops, mon_op_inactive_resources)) {
             continue;
-        }
 
-        if (!pe__rsc_running_on_any_node_in_list(rsc, only_show)) {
+        } else if (is_active && !pe__rsc_running_on_any_node_in_list(rsc, only_show)) {
```
Comment (kgaillot) [inline tools/crm_mon_print.c:210]: I think this should be is_partially_active, am I missing something?

Comment (clumens) [reply tools/crm_mon_print.c]: Nope, not missing anything.  I keep forgetting that partially_active is true for fully active, too.  Pushed a new patch set with just that change.
